internal/workflow.go

```diff
@@ -2497,6 +2521,15 @@ func WithRetryPolicy(ctx Context, retryPolicy RetryPolicy) Context {
 	return ctx1
 }
 
+// WithPriority adds priority to the copy of the context.
+//
+// Exposed as: [go.temporal.io/sdk/workflow.WithPriority]
+func WithPriority(ctx Context, priority Priority) Context {
+	ctx1 := setActivityParametersIfNotExist(ctx)
+	getActivityOptions(ctx1).Priority = convertToPBPriority(priority)
+	return ctx1
+}
```

Comment:
Same concern here, does this activity option need its own `With` helper? Added benefit of making people use `WithChildOptions` and `WithActivityOptions` to set this value is that it is not confusing which is for activity and which is for child (though I know that confusion already exists with some others like retry policy).

Reply:
We have these helpers for almost all options, there is a clear patterns that all `WithWorkflow` are for workflow and `With` are for activity if there is ambiguity